dvc/project.py

```diff
@@ -1179,7 +1179,7 @@ def graph(self, stages=None, from_directory=None):
 
         for stage in stages:
             for out in outs:
-                if stage.cwd == out.path:
+                if out.path in stage.cwd and not stage.is_data_source:
```

Comment:
The `not stage.is_data_source` is necessary to prevent detecting `dvc add --recursive directory` as "WorkingDirectoryAsOutput"

Reply:
`out.path in stage.cwd` doesn't work correctly when they simply have a common part of the path. E.g. `out.path == /path/to/something` and `stage.cwd == /path/to/something-1` would raise an error, even though it is perfectly valid.

 > The not stage.is_data_source is necessary to prevent detecting dvc add --recursive directory as "WorkingDirectoryAsOutput"

Could you elaborate? I can't see what that would be a problem.